File: src/test/java/com/alibaba/mtc/perf/memoryleak/NoMemoryLeak_ThreadLocal_NoRemove.java
package com.alibaba.mtc.perf.memoryleak;

import com.alibaba.mtc.perf.Utils;

/**
 * @author ding.lid
 */
public class NoMemoryLeak_ThreadLocal_NoRemove {
    public static void main(String[] args) throws Exception {
        long counter = 0;
        while (true) {
            ThreadLocal<String> threadLocal = new ThreadLocal<String>();
            threadLocal.set(Utils.getRandomString());

            if (counter % 1000 == 0)
                System.out.printf("%05dK\n", counter / 1000);
            counter++;
        }
    }
}

File: ttl2-compatible/src/test/java/com/alibaba/perf/memoryleak/NoMemoryLeak_ThreadLocal_NoRemove.kt
@file:JvmName("NoMemoryLeak_ThreadLocal_NoRemove")

package com.alibaba.perf.memoryleak

import com.alibaba.perf.getRandomString

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
fun main() {
    var counter: Long = 0
    while (true) {
        val threadLocal = ThreadLocal<String>()
        threadLocal.set(getRandomString())

        if (counter % 1000 == 0L)
            System.out.printf("%05dK%n", counter / 1000)
        counter++
    }
}
